File: leetcode3/FindSubarraysWithEqualSum/a.ts
// Copyright (C) 2022 by iamslash

// 75ms 96.67% 43.5MB 56.67%
// sliding window, set
// O(N) O(N)
function findSubarrays(nums: number[]): boolean {
    let sumSet = new Set<number>();
    for (let i = 1; i < nums.length; ++i) {
        let sum = nums[i-1] + nums[i];
        if (sumSet.has(sum)) {
            return true;
        }
        sumSet.add(sum);
    }
    return false;
};

File: leetcode3/FindSubarraysWithEqualSum/a.cpp
/* Copyright (C) 2022 by iamslash */

#include <cstdio>

using namespace std;

#include <vector>
#include <set>

// Wrong Answer For:
// [1, 2, 3]
// sliding window, set
// O(N) O(N)
class Solution {
public:
  bool findSubarrays(vector<int>& nums) {
    int n = nums.size();
    set<int> sumSet;
    for (int i = 1; i < n; ++i) {
      int sum = nums[i-1] + nums[i];
      if (sumSet.find(sum) != sumSet.end()) {
        return true;
      }
      sumSet.insert(sum);
    }
    return true;
  }
};

int main() {
  return 0;
}
